mava/utils/create_envs_utils.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Tuple
+
+import jumanji
+from jumanji.env import Environment
+from jumanji.environments.routing.lbf.generator import (
+    RandomGenerator as LbfRandomGenerator,
+)
+from jumanji.environments.routing.robot_warehouse.generator import (
+    RandomGenerator as RwareRandomGenerator,
+)
+from jumanji.wrappers import AutoResetWrapper
+
+from mava.wrappers.jumanji import LbfWrapper, RwareWrapper
+from mava.wrappers.shared import AgentIDWrapper, GlobalStateWrapper, LogWrapper
+
+# Registry mapping environment names to their generator and wrapper classes.
+JumanjiRegistry = {
+    "RobotWarehouse-v0": {"generator": RwareRandomGenerator, "wrapper": RwareWrapper},
+    "LevelBasedForaging-v0": {"generator": LbfRandomGenerator, "wrapper": LbfWrapper},
+}
+
+
+def create_jumanji_env(env_name: str, config: Dict) -> Tuple[Environment, Environment]:
+    """
+    Create a Jumanji environments for training and evaluation.
+    Args:
+        env_name (str): The name of the environment to create.
+        config (Dict): The configuration of the environment.
+    Returns:
+        A tuple of the environments.
+    """
+    # Config generator and select the wrapper.
+    generator = JumanjiRegistry[env_name]["generator"]
+    generator = generator(**config["env"]["scenario"]["task_config"])
+    wrapper = JumanjiRegistry[env_name]["wrapper"]
+
+    # Create envs.
+    env = jumanji.make(env_name, generator=generator)
+    env = wrapper(env)
+    eval_env = jumanji.make(env_name, generator=generator)
+    eval_env = wrapper(eval_env)
+
+    # Add agent id to observation.
+    if config["system"]["add_agent_id"]:
+        env = AgentIDWrapper(env)
+        eval_env = AgentIDWrapper(eval_env)
+
+    # Somehow it didn't work, only after switching the two ifs or do return GlobalStateWrapper(env)
+    if config["system"]["system_name"] in ["ff_mappo", "rec_mappo"]:
```

Comment:
I think system name isn't the way to do this, rather `add_global_state` so we don't have to check each system and then its also consistent with `add_agent_id`